frontend/explorer/public/js/hash.js

```diff
@@ -1209,25 +1209,100 @@ function populateHashPage(hash, explorerHash) {
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendTransactionStatistics(infoBody, explorerHash.transaction, explorerHash.unconfirmed!==true);
 		appendRawTransaction(infoBody, explorerHash.transaction.rawtransaction);
 	} else if (hashType === "unlockhash") {
+		appendNavigationMenuUnlockHash(hash);
 		appendHeading(infoBody, 'Hash Type: Unlock Hash');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendUnlockHashTables(infoBody, hash, explorerHash);
 	} else if (hashType === "coinoutputid") {
+		appendNavigationMenuCoinOutput(explorerHash, hash);
 		appendHeading(infoBody, 'Hash Type: Coin Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendCoinOutputTables(infoBody, hash, explorerHash);
 	} else if (hashType === "blockstakeoutputid") {
+		appendNavigationMenuBlockstakeOutput(explorerHash.transactions, hash);
 		appendHeading(infoBody, 'Hash Type: BlockStake Output ID');
 		appendHeading(infoBody, 'Hash: ' + hash);
 		appendBlockStakeOutputTables(infoBody, hash, explorerHash);
 	}
 }
 
+// appendNavigationMenuTranscaction adds the transaction link to the top navigation menu
+function appendNavigationMenuTransaction(explorerTransaction) {
+	appendNavigationMenuBlock(explorerTransaction);
+	var navigation = document.getElementById('nav-links');
+	var transactionSpan = document.createElement('span');
+	navigation.appendChild(document.createTextNode(' > '));
+	navigation.appendChild(transactionSpan);
+	linkHash(transactionSpan, explorerTransaction.id, 'Transaction');
+}
+
+// appendNavigationMenuCoinOuput adds the coin output link to the top navigation menu
+function appendNavigationMenuCoinOutput(explorerHash, hash) {
+	//Coin Ouput
+	if (explorerHash.transactions != null) {
+		for (var i = 0; i < explorerHash.transactions.length; i++) {
+			for (var j = 0; j < explorerHash.transactions[i].coinoutputids.length; j++) {
+				if (explorerHash.transactions[i].coinoutputids[j] == hash) {
+					appendNavigationMenuTransaction(explorerHash.transactions[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.transactions[i].coinoutputids[j], 'Coin Output');
+				}
+			} 
+		}
+	//Coin Ouput - Block Creator Reward
+	} else if (explorerHash.blocks != null) {
+		for (var i = 0; i < explorerHash.blocks.length; i++) {
+			for (var j = 0; j < explorerHash.blocks[i].minerpayoutids.length; j++) {
+				if (explorerHash.blocks[i].minerpayoutids[j] == hash) {
+					appendNavigationMenuBlock(explorerHash.blocks[i]);
+					var navigation = document.getElementById('nav-links');
+					var outputSpan = document.createElement('span');
+					navigation.appendChild(document.createTextNode(' > '));
+					navigation.appendChild(outputSpan);
+					linkHash(outputSpan, explorerHash.blocks[i].minerpayoutids[j], 'Payout ID');
```

Comment:
Not sure if payout ID is the most user-friendly name. Perhaps just call it Coin Output as well, given that it is a coin output anyhow.

Reply:
Changed